File: 3512_Minimum_Operations_to_Make_Array_Sum_Divisible_by_K/sol.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    // Strategy: sumofarray%k
    // Time:     O(N)
    // Space:    O(1)

    int solve(vector<int>& nums, int k) {
        int sum = 0;
        for(int x: nums){
            sum += x;
        }
        return sum%k;
    }
};

// 2. LOCAL TEST SUITE
struct TestCase {
    vector<int> input1;
    int input2; 
    int expected;
};

void run_tests() {
    Solution sol;
    vector<TestCase> testCases{
        {{3, 9, 7}, 5, 4}, 
        {{4, 1, 3}, 4, 0},
        {{3, 2}, 6, 5}
    };

    for(size_t i = 0; i < testCases.size(); ++i) {
        int result = sol.solve(testCases[i].input1, testCases[i].input2);
        bool isCorrect = (result == testCases[i].expected);
        cout << (isCorrect ? "✅" : "❌") << " Test #" << i;
        cout << " | Expected: " << testCases[i].expected 
             << " | Result: " << result << endl;
    }
}

int main() {
    run_tests();
    return 0;
}

File: 3512_Minimum_Operations_to_Make_Array_Sum_Divisible_by_K/sol.py
class Solution:
    """
    @param arg1: List
    @param arg2: int
    @return: int
    Time:    O(N)
    Space:   O(1)
    Strategy: sumofarray%k
    """
    def solve(self, nums: list[int] , k: int) -> int:
        return sum(nums)%k

# ================ 🧪 Local Test Suite ======================================
def run_tests():
    sol = Solution()
    test_cases = [
        (( [3,9,7], 5), 4),
        (( [4,1,3], 4) , 0),
        (( [3,2], 6) , 5),
    ]
    for i, (args, expected) in enumerate(test_cases, 1):
        result = sol.solve(*args)
        status = "✅ PASS" if result == expected else "❌ FAIL"
        print(f"Test {i}: | {status} | Input={args} | Expected={expected} | Got={result}")

if __name__ == "__main__":
    run_tests()
